src/main/java/teammates/common/util/FieldValidator.java

```diff
@@ -103,149 +73,107 @@
-    public static final String FEEDBACK_SESSION_NAME_FIELD_NAME = "feedback session name";
-    public static final int FEEDBACK_SESSION_NAME_MAX_LENGTH = 38;
-    public static final String FEEDBACK_SESSION_NAME_ERROR_MESSAGE =
-            "\"%s\" is not acceptable to TEAMMATES as " + FEEDBACK_SESSION_NAME_FIELD_NAME + " because it %s. "
-            + "The value of " + FEEDBACK_SESSION_NAME_FIELD_NAME + " should be no longer than "
-            + FEEDBACK_SESSION_NAME_MAX_LENGTH + " characters. It should not be empty.";
+    // TODO: allow smaller increments (currently only 1 hour increments allowed)
+    public static final String SESSION_START_TIME_FIELD_NAME = "start time";
+    public static final String SESSION_END_TIME_FIELD_NAME = "end time";
     
-    /*
-     * =======================================================================
-     * Field: Google ID
-     */
     public static final String GOOGLE_ID_FIELD_NAME = "Google ID";
     public static final int GOOGLE_ID_MAX_LENGTH = 254;
-    public static final String GOOGLE_ID_ERROR_MESSAGE =
-            "\"%s\" is not acceptable to TEAMMATES as a Google ID because it %s. "
-            + "A Google ID must be a valid id already registered with Google. "
-            + "It cannot be longer than " + GOOGLE_ID_MAX_LENGTH + " characters. "
-            + "It cannot be empty.";
     
-    /*
-     * =======================================================================
-     * Field: Gender
-     */
     public static final String GENDER_FIELD_NAME = "gender";
-    public static final List<String> GENDER_ACCEPTED_VALUES = Arrays.asList(Const.GenderTypes.MALE,
-                                                                            Const.GenderTypes.FEMALE,
-                                                                            Const.GenderTypes.OTHER);
+    public static final List<String> GENDER_ACCEPTED_VALUES =
+            Arrays.asList(Const.GenderTypes.MALE, Const.GenderTypes.FEMALE, Const.GenderTypes.OTHER);
+    
+    public static final String GIVER_TYPE_NAME = "feedback giver.";
+    public static final String RECIPIENT_TYPE_NAME = "feedback recipient.";
+    public static final String VIEWER_TYPE_NAME = "feedback viewer.";
+
+
+    ////////////////////
+    // ERROR MESSAGES //
+    ////////////////////
+
+    // possible reasons for invalidity
+    public static final String REASON_EMPTY = "is empty";
+    public static final String REASON_TOO_LONG = "is too long";
+    public static final String REASON_INCORRECT_FORMAT = "is not in the correct format";
+    public static final String REASON_CONTAINS_INVALID_CHAR = "contains invalid characters";
+    public static final String REASON_START_WITH_NON_ALPHANUMERIC_CHAR = "starts with a non-alphanumeric character";
+
+    // error message components
+    public static final String ERROR_INFO =
+            "\"{userInput}\" is not acceptable to TEAMMATES as {fieldName} because it {reason}.";
```

Comment:
By the way, the convention for placeholders is `${placeholder}` (missed the dollar sign).